spec/support/examples/shared_offline_transactions_examples.rb

```diff
@@ -0,0 +1,151 @@
+				.and_return(attributes)
+		end
+
+		it 'announces offline_transaction.created event' do
+			offline_transaction.publish_created
+
+			expect(event_publisher)
+				.to have_received(:announce)
+				.with(created_event_type, attributes)
+		end
+
+		it 'announces payment.created event' do
+			offline_transaction.publish_created
+
+			expect(event_publisher)
+				.to have_received(:announce)
+				.with(:payment_created, attributes)
+		end
+	end
+
+	describe '.publish_updated' do
+		let(:updated_event_name) { "#{off_trx_name}.updated" }
+		let(:updated_event_type) { updated_event_name.gsub('.', '_').to_sym }
+
+		let(:payment_updated_event) { double }
+		let(:offline_transaction_updated_event) { double }
+
+		before do
+			allow(offline_transaction)
+				.to receive(:to_event)
+				.with(updated_event_name, :nonprofit, :trx, :supporter, :subtransaction)
+				.and_return(offline_transaction_updated_event)
+
+			allow(offline_transaction_updated_event)
+				.to receive(:attributes!)
+				.and_return(attributes)
+
+			allow(offline_transaction)
+				.to receive(:to_event)
+				.with('payment.updated', :nonprofit, :trx, :supporter, :subtransaction)
+				.and_return(payment_updated_event)
+
+			allow(payment_updated_event)
+				.to receive(:attributes!)
+				.and_return(attributes)
+		end
+
+		it 'announces offline_transaction.updated event' do
+			offline_transaction.publish_updated
+
+			expect(event_publisher)
+				.to have_received(:announce)
+				.with(updated_event_type, attributes)
+		end
+
+		it 'announces payment.updated event' do
+			offline_transaction.publish_updated
+
+			expect(event_publisher)
+				.to have_received(:announce)
+				.with(:payment_updated, attributes)
+		end
+	end
+
+	describe '.publish_deleted' do
+		let(:deleted_event_name) { "#{off_trx_name}.deleted" }
+		let(:deleted_event_type) { deleted_event_name.gsub('.', '_').to_sym }
+
+		let(:payment_deleted_event) { double }
+		let(:offline_transaction_deleted_event) { double }
+
+		before do
+			allow(offline_transaction)
+				.to receive(:to_event)
+				.with(deleted_event_name, :nonprofit, :trx, :supporter, :subtransaction)
+				.and_return(offline_transaction_deleted_event)
+
+			allow(offline_transaction_deleted_event)
+				.to receive(:attributes!)
+				.and_return(attributes)
+
+			allow(offline_transaction)
+				.to receive(:to_event)
+				.with('payment.deleted', :nonprofit, :trx, :supporter, :subtransaction)
+				.and_return(payment_deleted_event)
+
+			allow(payment_deleted_event)
+				.to receive(:attributes!)
+				.and_return(attributes)
+		end
+
+		it 'announces offline_transaction.deleted event' do
+			offline_transaction.publish_deleted
+
+			expect(event_publisher)
+				.to have_received(:announce)
+				.with(deleted_event_type, attributes)
+		end
+
+		it 'announces payment.deleted event' do
+			offline_transaction.publish_deleted
+
+			expect(event_publisher)
+				.to have_received(:announce)
+				.with(:payment_deleted, attributes)
+		end
+	end
+end
```

Comment:
This is a good first pass! This stuff is not easy.

For the shared example, I want you to consider the situations where testing the results of OfflineTransactionRefund#publish_created might be useful: 

* Someone directly calls `OfflineTransactionRefund#publish_created`
* Someone calls `OfflineTransactionRefund.create` and we've added an `after_create` callback
* Someone calls an potential internal API inside the application like `ImaginaryOfflineTransactionProcessor.create_refund`
* Someone makes an HTTP call like `POST /api/transactions/v32134/refunds/`

In all of those cases, if we provide identical inputs, we know a matching `offline_transaction_refund.created` event will be published. Yes the HouIDs might be different, but we can use the `match_houid` method to match those. So the idea of a shared example is, given some sort of common interface in the spec description, the shared examples will run and we'll be able to verify whether these events were fired.